src/pages/_app.js

```diff
@@ -24,7 +24,7 @@ class MyApp extends React.Component {
     return (
       <Provider store={store}>
         <AppContainer>
-          <Component {...pageProps} key={router.route} />
+          <Component {...pageProps} router={router} key={router.route} />
```

Comment:
Consider using the [`withRouter` higher-order component](https://github.com/zeit/next.js#using-a-higher-order-component) - this is the more canonical way of accessing the router further down the component tree.